clusterctl/clusterdeployer/clientfactory.go

```diff
@@ -15,13 +15,22 @@ limitations under the License.
 */
 package clusterdeployer
 
-type clusterClientFactory struct {
+import (
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/cluster-api/pkg/clientcmd"
+)
+
+type clientFactory struct {
 }
 
-func NewClusterClientFactory() ClusterClientFactory {
-	return &clusterClientFactory{}
+func NewClientFactory() ClientFactory {
```

Comment:
I don't understand the point of this factory. It doesn't hold any state and just seems to wrap other public functions.